onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2006,10 +2020,29 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<Node*>& fuse
         }
       }
 
-      // Run TRT inference
-      if (!trt_context->enqueueV2(&buffers[0], stream, nullptr)) {
-        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "TensorRT EP execution context enqueue failed.");
-      }
+      // Run TRT inference			  
+      if (trt_state->cuda_graph_enable)
+      {
+          if (*cuda_graph == nullptr) {
+            cudaGraph_t graph;
+            *cuda_graph = &(trt_state->cuda_graph_instance);
+            //warm up for cuda graph capturing
+            if (!trt_context->enqueueV2(&buffers[0], stream, nullptr)) {
+              return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "TensorRT EP execution context enqueue failed.");
+            }   
+            cudaStreamBeginCapture(stream, cudaStreamCaptureModeRelaxed );
```

Comment:
Should the CUDA calls have CUDA_CALL_THROW() to deal with CUDA call errors if any ?